### PR TITLE
Upgrade from Chromium 107 to Chromium 108.

### PR DESCRIPTION
**Pull request by mkarolin**
Resolves https://github.com/brave/brave-browser/issues/25703

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by wchen342 — APPROVED** [review]
Android test change lgtm

**Review by simonhong — APPROVED** [review]
`browser/widevine`: ++

**Review by samartnik — APPROVED** [review]
Java++

**Review by darkdh — APPROVED** [review]
sync and wallet-core ++

File: ios/browser/api/ads/BUILD.gn
Comment (kylehickinson) [inline ios/browser/api/ads/BUILD.gn:16]: Was this a warning turned error in the iOS wrapper specifically? We should probably open a separate issue for after this merges to just fix the warning properly and remove shadowing

**Review by kylehickinson — APPROVED** [review]
iOS++, I'll open up a separate issue to remove all new configs that disable the no-shadow rule

**Review by fmarier — APPROVED** [review]
`script/brave_license_helper.py` changes look good.

File: chromium_src/chrome/browser/ui/views/tabs/tab_hover_card_test_util.cc
Comment (bridiver) [inline chromium_src/chrome/browser/ui/views/tabs/tab_hover_card_test_util.cc:6]: it's not clear why this change was needed

File: chromium_src/third_party/blink/renderer/core/script/script_loader.cc
Comment (bridiver) [inline chromium_src/third_party/blink/renderer/core/script/script_loader.cc:16]: where is this used?

File: chromium_src/chrome/browser/privacy_sandbox/privacy_sandbox_service_unittest.cc
Comment (bridiver) [inline chromium_src/chrome/browser/privacy_sandbox/privacy_sandbox_service_unittest.cc:162]: not sure I'm following what's going on in these tests. The privacy sandbox is disabled, right? cc @pes10k

File: chromium_src/services/network/cookie_settings.cc
Comment (bridiver) [inline chromium_src/services/network/cookie_settings.cc:79]: these should be empty, right?

File: ios/browser/api/certificate/models/brave_certificate_signature.mm
Comment (bridiver) [inline ios/browser/api/certificate/models/brave_certificate_signature.mm:27]: shouldn't we be checking has_value here if these are optional?

File: patches/third_party-blink-common-permissions-permission_utils.cc.patch
Comment (bridiver) [inline patches/third_party-blink-common-permissions-permission_utils.cc.patch:9]: `#define NUM .... NUM`

File: patches/chrome-BUILD.gn.patch
Comment (bridiver) [inline patches/chrome-BUILD.gn.patch:60]: we should also use `include_branded_entitlements` to selectively exclude our entitlements for locally signed builds

File: patches/chrome-installer-util-BUILD.gn.patch
Comment (bridiver) [inline patches/chrome-installer-util-BUILD.gn.patch:17]: This patch is a mess. We should be setting branding_path_component in config.js, not here. We should also remove the comments and do a follow-up ticket because the brave_strings.grd thing is wrong as well, but requires more work to fix. Another patch is causing this to be a problem
```
 grit("chromium_strings") {
-  source = "chromium_strings.grd"
+  source = "${branding_path_component}_strings.grd"
```
we should just be using chromium_strings.grd instead of patching branding_path_component and this

File: chromium_src/services/network/cookie_settings.cc
Comment (bridiver) [reply chromium_src/services/network/cookie_settings.cc:79]: just verifying that this stuff is disabled

File: chromium_src/chrome/common/url_constants.cc
Comment (iefremov) [inline chromium_src/chrome/common/url_constants.cc:26]: i would append trailing `/` here and above just for consistency :D

**Review by zenparsing — APPROVED** [review]
Rewards 👍

File: chromium_src/chrome/browser/ui/views/tabs/tab_hover_card_test_util.cc
Comment (mkarolin) [reply chromium_src/chrome/browser/ui/views/tabs/tab_hover_card_test_util.cc:6]: This is needed because of this upstream change https://source.chromium.org/chromium/chromium/src/+/4021762e0a866da98e41e9813c628cd8793aa84c - it added 
<pre>
TabHoverCardController* hover_card_controller_for_testing() {
    return hover_card_controller_.get();
}
</pre>

to `TabStrip` class, which is used in this file. But, we use `chromium_src` override on `TabStrip` to replace `TabHoverCardController` with `BraveTabHoverCardController`, so we need to add the header to avoid the `member access into incomplete type` error.

File: chromium_src/chrome/browser/ui/views/permissions/chip_controller.cc
Comment (mkarolin) [reply chromium_src/chrome/browser/ui/views/permissions/chip_controller.cc:7]: This is related to https://source.chromium.org/chromium/chromium/src/+/eeb6206faf566e6b73f2a62c7671e6b2aca7434b. This happens because we make additional calls to `LocationBarView::Update` from `BraveLocationBarView::Init` (via `Update` call) and then in `OnThemeChanged` while `GetLocationBarView()` still returns `nullptr`.

File: chromium_src/chrome/common/url_constants.cc
Comment (mkarolin) [reply chromium_src/chrome/common/url_constants.cc]: Done via 420ff1e07c

File: patches/third_party-blink-common-permissions-permission_utils.cc.patch
Comment (mkarolin) [reply patches/third_party-blink-common-permissions-permission_utils.cc.patch:9]: Sadly, `NUM` is used in the same file later: 
<pre>
const int NUM_TYPES = static_cast<int>(PermissionType::NUM);
</pre>

**Review by onyb — APPROVED** [review]
wallet frontend ++

File: ios/browser/api/ads/BUILD.gn
Comment (cdesouza-chromium) [reply ios/browser/api/ads/BUILD.gn]: Fixed in 08a03ba673c7169836c91314e3d28087764661eb.

File: ios/browser/api/certificate/models/brave_certificate_signature.mm
Comment (cdesouza-chromium) [reply ios/browser/api/certificate/models/brave_certificate_signature.mm:27]: @kylehickinson has raised this above in another comment. I'm waiting for a reply there.

File: ios/browser/api/ledger/BUILD.gn
Comment (cdesouza-chromium) [reply ios/browser/api/ledger/BUILD.gn]: Fixed in 08a03ba673c7169836c91314e3d28087764661eb

File: patches/chrome-installer-util-BUILD.gn.patch
Comment (mkarolin) [reply patches/chrome-installer-util-BUILD.gn.patch]: Simplified slightly in [32de785](https://github.com/brave/brave-core/pull/15294/commits/32de78500e975e82b295ed319687d07a1503dff5). I left the comment in for now so that the info isn't lost for the follow up. There's more involved with brave_strings.grd than just these 2 patches.

File: ios/browser/api/certificate/models/brave_certificate_signature.mm
Comment (kylehickinson) [reply ios/browser/api/certificate/models/brave_certificate_signature.mm]: @mkarolin Unfortunately that fix introduces a potential crash as well because those properties are marked `nonnull` by default. Swift will import them as such and if there's a nullptr under it–which would happen if `has_value` returns false–it will crash.  We have 2 options here:

- Make the `digest` and `algorithm` properties nullable (i.e in `brave_certificate_signature.h`, doing `@property(nonatomic, readonly, nullable) ...` on the two properties)
- Give the `digest` and `algorithm` default values such as empty strings.

I'd prefer the former, and we can deal with the optionals in `brave-ios`.

File: chromium_src/services/network/cookie_settings.cc
Comment (mkarolin) [reply chromium_src/services/network/cookie_settings.cc:79]: yes, we have `kFirstPartySets` feature disabled in [chromium_src/content/public/common/content_features.cc](https://github.com/brave/brave-core/blob/master/chromium_src/content/public/common/content_features.cc#L17)

File: chromium_src/chrome/browser/privacy_sandbox/privacy_sandbox_service_unittest.cc
Comment (mkarolin) [reply chromium_src/chrome/browser/privacy_sandbox/privacy_sandbox_service_unittest.cc:162]: Yes, privacy sandbox is disabled in https://github.com/brave/brave-core/blob/master/chromium_src/content/public/common/content_features.cc#L21 and in https://github.com/brave/brave-core/blob/master/chromium_src/components/privacy_sandbox/privacy_sandbox_features.cc

File: patches/chrome-BUILD.gn.patch
Comment (mkarolin) [reply patches/chrome-BUILD.gn.patch:60]: Is this for a follow up?

File: chromium_src/chrome/browser/ui/views/permissions/chip_controller.cc
Comment (bridiver) [reply chromium_src/chrome/browser/ui/views/permissions/chip_controller.cc:7]: I feel like we're doing something wrong here if this is happening. Can you create a follow-up issue to look at it and maybe @petemill can check it out?

File: chromium_src/chrome/browser/privacy_sandbox/privacy_sandbox_service_unittest.cc
Comment (bridiver) [reply chromium_src/chrome/browser/privacy_sandbox/privacy_sandbox_service_unittest.cc:162]: so if fps is also disabled, what is this file doing? I don't even see any actual tests here

File: chromium_src/chrome/browser/ui/views/tabs/tab_hover_card_test_util.cc
Comment (bridiver) [reply chromium_src/chrome/browser/ui/views/tabs/tab_hover_card_test_util.cc:6]: can you add comments to the file explaining?

File: patches/chrome-BUILD.gn.patch
Comment (bridiver) [reply patches/chrome-BUILD.gn.patch:60]: yes

File: patches/chrome-installer-util-BUILD.gn.patch
Comment (mkarolin) [reply patches/chrome-installer-util-BUILD.gn.patch]: Based on DM, additional fix via https://github.com/brave/brave-core/pull/15294/commits/43759de1477baab7fd8dc59d90f1f2f468783c8d

File: patches/chrome-BUILD.gn.patch
Comment (mkarolin) [reply patches/chrome-BUILD.gn.patch:60]: Filed https://github.com/brave/brave-browser/issues/26651

File: ios/browser/api/certificate/models/brave_certificate_signature.mm
Comment (mkarolin) [reply ios/browser/api/certificate/models/brave_certificate_signature.mm]: @kylehickinson wouldn't 
<pre>
    _algorithm = [[NSString alloc] init];
    _digest = [[NSString alloc] init];
</pre>
have initialize them to empty strings?

File: chromium_src/chrome/browser/ui/views/permissions/chip_controller.cc
Comment (mkarolin) [reply chromium_src/chrome/browser/ui/views/permissions/chip_controller.cc:7]: Issue created https://github.com/brave/brave-browser/issues/26653

File: chromium_src/chrome/browser/privacy_sandbox/privacy_sandbox_service_unittest.cc
Comment (mkarolin) [reply chromium_src/chrome/browser/privacy_sandbox/privacy_sandbox_service_unittest.cc:162]: Hmm, it seems Mario removed the tests as no longer needed in [Chromium 101 bump](https://github.com/brave/brave-core/commit/383544c8a52e150997076600bacc151b873d4217), but left the file around for some reason. We can just remove it, I think.

Comment (mkarolin) [reply chromium_src/chrome/browser/privacy_sandbox/privacy_sandbox_service_unittest.cc]: Removed in c0dc9729faa983e1f51f6f2c1510295694284e51

File: chromium_src/chrome/browser/ui/views/tabs/tab_hover_card_test_util.cc
Comment (mkarolin) [reply chromium_src/chrome/browser/ui/views/tabs/tab_hover_card_test_util.cc:13]: Added in 859fbd38787c530022732af7c10688e0a32b95e1

File: patches/chrome-browser-ui-views-location_bar-location_bar_view.h.patch
Comment (goodov) [inline patches/chrome-browser-ui-views-location_bar-location_bar_view.h.patch:25]: should be located before `private:`

File: patches/components-omnibox-browser-autocomplete_controller.cc.patch
Comment (goodov) [inline patches/components-omnibox-browser-autocomplete_controller.cc.patch]: unnecessary empty line

File: patches/third_party-blink-renderer-modules-mediastream-media_devices.cc.patch
Comment (goodov) [inline patches/third_party-blink-renderer-modules-mediastream-media_devices.cc.patch:10]: unnecessary empty line.

File: chromium_src/storage/browser/blob/blob_url_store_impl.cc
Comment (goodov) [inline chromium_src/storage/browser/blob/blob_url_store_impl.cc:124]: `const url::Origin& storage_key_origin = storage_key_.origin();`

File: patches/chrome-browser-ui-views-location_bar-location_bar_view.h.patch
Comment (mkarolin) [reply patches/chrome-browser-ui-views-location_bar-location_bar_view.h.patch:25]: I think it is? This is inserted on line 253, `private:` is located on line 265.

File: patches/components-omnibox-browser-autocomplete_controller.cc.patch
Comment (mkarolin) [reply patches/components-omnibox-browser-autocomplete_controller.cc.patch]: This was fixed in cb3330c4e6

File: patches/third_party-blink-renderer-modules-mediastream-media_devices.cc.patch
Comment (mkarolin) [reply patches/third_party-blink-renderer-modules-mediastream-media_devices.cc.patch]: Fixed in 4c3223079e

File: ios/browser/api/certificate/models/brave_certificate_signature.mm
Comment (kylehickinson) [reply ios/browser/api/certificate/models/brave_certificate_signature.mm]: @mkarolin yeah that would do it, sorry I'm blind! Carry on... 🙂

File: chromium_src/storage/browser/blob/blob_url_store_impl.cc
Comment (mkarolin) [reply chromium_src/storage/browser/blob/blob_url_store_impl.cc]: Fixed in 4b99374e34

**Review by goodov — APPROVED** [review]

**Review by iefremov — APPROVED** [review]

**Comment by mkarolin** [comment]
Rebased on master and squashed fixups.

**Review by diracdeltas — APPROVED** [review]

**Comment by mkarolin** [comment]
Encountered https://github.com/brave/brave-browser/issues/25672 with the QA test build for Win x64 (https://ci.brave.com/view/linux/job/test-brave-browser-build-windows-x64/674/execution/node/121/log/). Adjusted the fix to apply to x64 win builds as well (but restricted it to Release only since there were no issues with either Debug or Static builds). cc: @bridiver @goodov

**Comment by mkarolin** [comment]
2 Chromium browser tests failed on Windows x86 only in the latest CI run. Will follow up on these separately: 
<pre>
 2 tests failed:
     CDM_10/ECKEncryptedMediaTest.MultipeCdmTypes/0 (..\..\chrome\browser\media\encrypted_media_browsertest.cc:957)
     CDM_11/ECKEncryptedMediaTest.MultipeCdmTypes/0 (..\..\chrome\browser\media\encrypted_media_browsertest.cc:957)
</pre>

**Comment by mkarolin** [comment]
When uplifting, also uplift:
* https://github.com/brave/brave-browser/issues/26783
* https://github.com/brave/brave-browser/issues/26805
* https://github.com/brave/brave-browser/issues/26799

**Pull request merged**
